net: ip: net_if: bypassing IP stack with SOCK_RAW sockets

jhirsi opened the pull request:
Continuation to https://github.com/zephyrproject-rtos/zephyr/pull/54664

Bypass the IP stack and go directly to L2 with SOCK_RAW/IPPROTO_RAW sockets in net_if_send_data().

jhirsi commented:
Just rebased (no actual chenged compared the original PR) because there waskind of a complaint of it. But seems that it removed the approvals, apologies about hashlings, so can you please recheck. @rlubos / @jukkar

jukkar commented on subsys/net/ip/net_if.c:
```diff
@@ -481,6 +481,13 @@ enum net_verdict net_if_send_data(struct net_if *iface, struct net_pkt *pkt)
 	}
```
LGTM, could you change the commit subject to `net: iface: ....` as the first part is not meant to be a file/directory but a logical component that the commit is changing.

jhirsi replied:
done.

jhirsi committed:
net: iface: bypassing IP stack with SOCK_RAW sockets

Bypass the IP stack and go directly to L2 with
SOCK_RAW/IPPROTO_RAW sockets in net_if_send_data().

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>
